app/grandchallenge/components/form_fields.py

```diff
@@ -102,10 +105,27 @@ def get_initial_value(self):
             return self.initial
 
     def get_image_field(self):
+        current_value = None
+
+        if isinstance(self.initial, ComponentInterfaceValue):
+            current_value = self.initial.image
+        elif self.initial:
+            try:
+                uuid = UUID(self.initial, version=4)
+            except ValueError:
+                uuid = None
+
```

Comment:
Why do you need this? Can't you just check `Image.objects.filter(pk=self.initial).exists()` and `UserUpload.objects.filter(pk=self.initial).exists()` directly? Seems like an unnecessary conversion step here.

Reply:
before I implemented the check `if self.initial`, there was the case when submitting without choosing an image from the image search widget that causes a 500 error failing on `Image.objects.filter(pk=self.initial).exists()` because the submitted value is not a valid UUID (empty string), which also means if any other value submitted (like the screenshot below) will cause a 500 error. Currently, after the check `if self.initial` that case does not happen anymore, but still possible by tampering the form variables.
That is why I thought of checking for a valid UUID first before doing `filter()`.
Is that over protective? should I remove it?

<img width="1685" alt="Screenshot 2024-11-06 at 09 22 00" src="https://github.com/user-attachments/assets/990f70f5-5fa7-4c9a-8f3e-d801026f7082">

Reply:
Ah ok. How was it possible to submit without a chosen image for the ImageSelect widget? I would expect there to be a user error "This field is required" that prevent the user from submitting without a value. 

And as for the possibility of tinkering with the form data, ok, but in that case all we want is to be made aware of this happening, which will happen either way (whether you convert to uuid first and then raise a RuntimeError or whether you let it fail at the query), so I think it is not needed. 

And jut to make sure, if an error is raised here, the form won't be submitted, right? I'm asking because we don't want algorithm jobs stuck in strange states.

Reply:
There was no check for initial value since the value was not captured on form redisplay (that is the main issue), and thus, current_value was None by default.
I fixed it as proposed and it works as intended now.

Reply:
Yes, indeed, I understand how the current_value was None on resubmission, but that should still have triggered a ValidationError on the form and prevent submitting. This is what happens on main when I try it out, so I'm trying to understand what happened in your case.

Reply:
That is my point, because `current_value` was None by default and not set to the value of the form element on submission, which in the case of not choosing an image, will be an empty string sent with the form and leading to that exception when trying to check if it is a `pk` for `Image` or `UserUpload`. So now, we need to check that initial is not empty or None before trying to check if it is a `pk` for an `Image` or `UserUpload`.

Reply:
Ok, clear.